### PR TITLE
settings redesign: Use a single generic passdb/userdb_fields

### PR DESCRIPTION
**Pull request by sirainen**
DOV-6485

### CODE REVIEW

File: source/configuration_manual/authentication/password_databases_passdb.rst [hunk 1]
```diff
@@ -170,6 +170,25 @@ Passdb setting
...
 
+.. dovecot_core:setting:: passdb_fields
+   :values: @strlist
+
+   Passdb fields (and :ref:`authentication-password_database_extra_fields`).
+   The values can contain :ref:`%variables <config_variables>`. All %variables
+   used here reflect the state **after** the current passdb lookup, and can
+   refer to fields returned by previous passdb lookups. Depending on the passdb
+   driver, it can also refer to variable fields returned by it (e.g.
+   ``%{ldap:fieldName}``).
```
Comment (cmouse) [inline source/configuration_manual/authentication/password_databases_passdb.rst:181]: do we support setting arbitrary fields like

```
passdb pam {
   driver = pam
   fields = {
      brand = iron
  }
}
```

Comment (sirainen) [reply source/configuration_manual/authentication/password_databases_passdb.rst]: added some examples

File: source/configuration_manual/authentication/password_databases_passdb.rst [hunk 2]
```diff
@@ -170,6 +170,35 @@ Passdb setting
...
+   Passdb fields (and :ref:`authentication-password_database_extra_fields`).
+   The values can contain :ref:`%variables <config_variables>`. All %variables
+   used here reflect the state **after** the current passdb lookup, and can
+   refer to fields returned by previous passdb lookups. Depending on the passdb
+   driver, it can also refer to variable fields returned by it (e.g.
+   ``%{ldap:fieldName}``). Example:
+
+   .. code-block:: none
+
+      passdb ldap1 {
```
Comment (cmouse) [inline source/configuration_manual/authentication/password_databases_passdb.rst:185]: should use passdb ldap here instead? or do we want two examples of implicit and explicit driver?

Comment (sirainen) [reply source/configuration_manual/authentication/password_databases_passdb.rst:185]: This is now getting mixed with changes in https://github.com/dovecot/documentation/pull/853. I suppose I could already do the final change here.

**Review by cmouse — APPROVED** [review]

**Pull request merged**